### PR TITLE
feat: introduce cgo free sqlite

### PR DESCRIPTION
**Pull request by Yisaer**
Introduce https://github.com/logoove/sqlite as the cgo-free sqlite which based on the https://gitlab.com/cznic/sqlite with several convenient fixing.

### CODE REVIEW

File: internal/pkg/store/sql/sqlite/cgo_free_sqllite.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
+//go:build !cgo
```
Comment (ngjaying) [inline internal/pkg/store/sql/sqlite/cgo_free_sqllite.go:1]: I was trying to replace the old driver. Are there any reason to keep the cgo version?

File: internal/pkg/store/sql/sqlite/cgo_free_sqllite.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	"fmt"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/lf-edge/ekuiper/internal/conf"
+	"github.com/lf-edge/ekuiper/internal/pkg/store/definition"
+
+	// introduce sqlite
+	_ "github.com/logoove/sqlite"
```
Comment (ngjaying) [inline internal/pkg/store/sql/sqlite/cgo_free_sqllite.go:32]: Why using this?

File: go.mod [hunk 3]
```diff
@@ -22,6 +22,7 @@ require (
 	github.com/keepeye/logrus-filename v0.0.0-20190711075016-ce01a4391dd1
 	github.com/klauspost/compress v1.15.11
 	github.com/lestrrat-go/file-rotatelogs v2.4.0+incompatible
+	github.com/logoove/sqlite v1.15.3
 	github.com/mattn/go-sqlite3 v1.14.16
```
Comment (ngjaying) [inline go.mod:26]: Aim to remove mattn/go-sqlite dependency

File: internal/pkg/store/sql/sqlite/cgo_free_sqllite.go (hunk 2 above)
Comment (Yisaer) [reply internal/pkg/store/sql/sqlite/cgo_free_sqllite.go]: `czniz/sqlite` use `sqlite` as driver name, which is different from the original sqlite driver and will cause problems if we use `dburl` to parse url. This one forked the `czniz/sqlite` and fixed this problem.

File: internal/pkg/store/sql/sqlite/database.go [hunk 4]
```diff
@@ -1,10 +1,10 @@
-// Copyright 2022-2022 EMQ Technologies Co., Ltd.
+// Copyright 2023-2023 EMQ Technologies Co., Ltd.
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+// http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ngjaying) [inline internal/pkg/store/sql/sqlite/database.go:7]: Here, can we revert? Otherwise, this will cause meaningless diff if others change this file.

**Review by ngjaying — CHANGES_REQUESTED** [review]
Some small problem in copyright statement

**Review by ngjaying — APPROVED** [review]
cool, LGTM

**Pull request merged**